File: projects/Ляшук/lab7/app/src/main/java/com/example/lab7/AudioActivity.java
package com.example.lab7;

import android.media.MediaPlayer;
import android.os.Bundle;
import android.os.Handler;
import android.widget.Button;
import android.widget.SeekBar;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

public class AudioActivity extends AppCompatActivity {
    private MediaPlayer mediaPlayer;
    private SeekBar seekBar;
    private Handler handler = new Handler();
    private DatabaseHelper dbHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_audio);

        dbHelper = new DatabaseHelper(this);
        dbHelper.addRecord("Открыта активность Аудио (Ляшук В.И.)");

        try {
            int resId = getResources().getIdentifier("audio_sample", "raw", getPackageName());
            if (resId != 0) {
                mediaPlayer = MediaPlayer.create(this, resId);
            } else {
                Toast.makeText(this, "Файл audio_sample не найден в res/raw!", Toast.LENGTH_LONG).show();
            }
        } catch (Exception e) {
            e.printStackTrace();
        }

        seekBar = findViewById(R.id.audioSeekBar);
        Button btnPlay = findViewById(R.id.btnPlay);
        Button btnPause = findViewById(R.id.btnPause);
        Button btnStop = findViewById(R.id.btnStop);
        Button btnBack = findViewById(R.id.btnBackAudio);

        btnPlay.setOnClickListener(v -> {
            if (mediaPlayer != null) {
                mediaPlayer.start();
                updateSeekBar();
                Toast.makeText(this, "Ляшук В.И.: Воспроизведение", Toast.LENGTH_SHORT).show();
                dbHelper.addRecord("Воспроизведение аудио начато (Ляшук В.И.)");
            }
        });

        btnPause.setOnClickListener(v -> {
            if (mediaPlayer != null && mediaPlayer.isPlaying()) {
                mediaPlayer.pause();
                dbHelper.addRecord("Аудио на паузе");
            }
        });

        btnStop.setOnClickListener(v -> {
            if (mediaPlayer != null) {
                mediaPlayer.stop();
                try {
                    mediaPlayer.prepare();
                    mediaPlayer.seekTo(0);
                } catch (Exception e) { e.printStackTrace(); }
                dbHelper.addRecord("Аудио остановлено");
            }
        });

        btnBack.setOnClickListener(v -> finish());
    }

    private void updateSeekBar() {
        if (mediaPlayer != null) {
            seekBar.setMax(mediaPlayer.getDuration());
            handler.postDelayed(new Runnable() {
                @Override
                public void run() {
                    if (mediaPlayer != null) {
                        seekBar.setProgress(mediaPlayer.getCurrentPosition());
                        handler.postDelayed(this, 1000);
                    }
                }
            }, 0);
        }
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mediaPlayer != null) {
            mediaPlayer.release();
            mediaPlayer = null;
        }
    }
}

File: projects/Гончеренок/lab5/src/MediaApp/app/src/main/java/com/example/mediaapp/AudioActivity.kt
package com.example.mediaapp

import android.media.MediaPlayer
import android.os.Bundle
import android.widget.Button
import androidx.appcompat.app.AppCompatActivity

class AudioActivity : AppCompatActivity() {

    private var mediaPlayer: MediaPlayer? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_audio)

        title = "Audio — Гончерёнок К.А. АС-66"

        val btnPlay = findViewById<Button>(R.id.btnPlayAudio)
        val btnPause = findViewById<Button>(R.id.btnPauseAudio)
        val btnStop = findViewById<Button>(R.id.btnStopAudio)
        val btnBack = findViewById<Button>(R.id.btnBackAudio)

        mediaPlayer = MediaPlayer.create(this, R.raw.my_audio)

        btnPlay.setOnClickListener {
            mediaPlayer?.start()
        }

        btnPause.setOnClickListener {
            if (mediaPlayer?.isPlaying == true) {
                mediaPlayer?.pause()
            }
        }

        btnStop.setOnClickListener {
            mediaPlayer?.release()
            mediaPlayer = MediaPlayer.create(this, R.raw.my_audio)
        }

        btnBack.setOnClickListener {
            finish()
        }
    }

    override fun onDestroy() {
        mediaPlayer?.release()
        mediaPlayer = null
        super.onDestroy()
    }
}